contracts/protocol/pool/PoolApeStaking.sol

```diff
@@ -188,30 +180,33 @@ contract PoolApeStaking is
         DataTypes.PoolStorage storage ps = poolStorage();
         checkSApeIsNotPaused(ps);
 
-        DataTypes.ReserveData storage nftReserve = ps._reserves[nftAsset];
-        address xTokenAddress = nftReserve.xTokenAddress;
-        INTokenApeStaking nTokenApeStaking = INTokenApeStaking(xTokenAddress);
-        IERC721 bakcContract = nTokenApeStaking.getBAKC();
+        address xTokenAddress = ps._reserves[nftAsset].xTokenAddress;
+        IERC721 bakcContract = INTokenApeStaking(xTokenAddress).getBAKC();
+        address bakcNToken = ps._reserves[address(bakcContract)].xTokenAddress;
+        address[] memory originArray = new address[](_nftPairs.length);
         for (uint256 index = 0; index < _nftPairs.length; index++) {
             require(
                 INToken(xTokenAddress).ownerOf(_nftPairs[index].mainTokenId) ==
                     msg.sender,
                 Errors.NOT_THE_OWNER
             );
-            bakcContract.safeTransferFrom(
-                msg.sender,
+
+            originArray[index] = validateBAKCOwnerAndTransfer(
+                bakcContract,
+                _nftPairs[index].bakcTokenId,
+                bakcNToken,
                 xTokenAddress,
-                _nftPairs[index].bakcTokenId
+                msg.sender
             );
         }
 
-        nTokenApeStaking.claimBAKC(_nftPairs, msg.sender);
+        INTokenApeStaking(xTokenAddress).claimBAKC(_nftPairs, msg.sender);
```

Comment:
The cost is very little. and if we have a stack variable to store this value, it might cause stack too deep, then it'll cost more gas to use memory struct.